vendor/github.com/openshift/library-go/test/library/etcdverticalscaling/helpers.go

```diff
@@ -0,0 +1,282 @@
+func EnsureMembersCount(t TestingT, etcdClientFactory EtcdClientCreator, expectedMembersCount int) error {
+	waitPollInterval := 15 * time.Second
+	waitPollTimeout := 10 * time.Minute
+	t.Logf("Waiting up to %s for the cluster to reach the expected member count of %v", waitPollTimeout.String(), expectedMembersCount)
+
+	if err := wait.Poll(waitPollInterval, waitPollTimeout, func() (bool, error) {
+		etcdClient, closeFn, err := etcdClientFactory.NewEtcdClient()
+		if err != nil {
+			t.Logf("failed to get etcd client, will retry, err: %v", err)
+			return false, nil
+		}
+		defer closeFn()
+		defer etcdClient.Close() // ignore the errors
+
+		ctx, cancel := context.WithTimeout(context.TODO(), 15*time.Second)
+		defer cancel()
+		memberList, err := etcdClient.MemberList(ctx)
+		if err != nil {
+			t.Logf("failed to get the member list, will retry, err: %v", err)
+			return false, nil
+		}
+
+		var memberNames []string
+		for _, member := range memberList.Members {
+			memberNames = append(memberNames, member.Name)
+		}
+		if len(memberNames) != expectedMembersCount {
+			t.Logf("unexpected number of etcd members, expected exactly %d, got: %v, current members are: %v", expectedMembersCount, len(memberNames), memberNames)
+			return false, nil
+		}
+
+		t.Logf("cluster have reached the expected number of %v members, the members are: %v", expectedMembersCount, memberNames)
+		return true, nil
+	}); err != nil {
+		newErr := fmt.Errorf("failed on waiting for the cluster to reach the expected member count of %v, err %v", expectedMembersCount, err)
+		return newErr
+	}
+	return nil
+}
+
+func EnsureMemberRemoved(etcdClientFactory EtcdClientCreator, memberName string) error {
+	etcdClient, closeFn, err := etcdClientFactory.NewEtcdClient()
+	if err != nil {
+		return err
+	}
+	defer closeFn()
+	defer etcdClient.Close() // ignore the errors
+
+	ctx, cancel := context.WithTimeout(context.TODO(), 15*time.Second)
+	defer cancel()
+	rsp, err := etcdClient.MemberList(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, member := range rsp.Members {
+		if member.Name == memberName {
+			return fmt.Errorf("member %v hasn't been removed", spew.Sdump(member))
+		}
+	}
+	return nil
+}
+
+func EnsureHealthyMember(t TestingT, etcdClientFactory EtcdClientCreator, memberName string) error {
+	etcdClient, closeFn, err := etcdClientFactory.NewEtcdClientForMember(memberName)
+	if err != nil {
+		return err
+	}
+	defer closeFn()
+	defer etcdClient.Close() // ignore the errors
+
+	// since we have a direct connection with the member
+	// getting any response is a good sign of healthiness
+	ctx, cancel := context.WithTimeout(context.TODO(), 15*time.Second)
+	defer cancel()
+	_, err = etcdClient.Get(ctx, "health")
+	if err != nil {
+		return fmt.Errorf("failed to check healthiness condition of the %q member, err: %v", memberName, err)
+	}
```

Comment:
This won't work if the request gets sent to a learner member which doesn't support linearized read/Get.
We have to use a serialized request for that case.
See https://github.com/openshift/cluster-etcd-operator/pull/758/files#diff-369b1a2393ad554f2b3caef6751bf1c596b99167ffacc57a3e95d2bf3e9f7f68R164-R170
```Go
if member.IsLearner {
	// Serializable (local) read request for learner members
	// since learners don't support linearized reads
	_, err = etcdClient.Get(ctx, "health", clientv3.WithSerializable())
} else {
	// Linearized read request to verify health of voting members
	_, err = etcdClient.Get(ctx, "health")
}
```

Reply:
@hasbro17 I suggest we should change the test first in your PR (https://github.com/openshift/cluster-etcd-operator/pull/758) otherwise we won't be able to merge it.